arangod/GeneralServer/GeneralServer.cpp

```diff
@@ -185,12 +185,28 @@ IoContext& GeneralServer::selectIoContext() {
   return _contexts[lowpos];
 }
 
-asio_ns::ssl::context& GeneralServer::sslContext() {
+std::shared_ptr<asio_ns::ssl::context> GeneralServer::sslContext() {
   std::lock_guard<std::mutex> guard(_sslContextMutex);
   if (!_sslContext) {
     _sslContext.reset(new asio_ns::ssl::context(SslServerFeature::SSL->createSslContext()));
   }
-  return *_sslContext;
+  return _sslContext;
+}
+
+Result GeneralServer::reloadTLS() {
+  std::lock_guard<std::mutex> guard(_sslContextMutex);
+  try {
+    _sslContext.reset(new asio_ns::ssl::context(SslServerFeature::SSL->createSslContext()));
```

Comment:
doesn't this copy the context at least once

Reply:
Right. I changed the helper to return shared_ptrs right away.